File: java/sort/array/BubbleSort.java
package sort.array;

public class BubbleSort {
    public void sort(int[] nums) {
        int border = nums.length - 1;
        for (int _ : nums) {
            boolean flag = false;
            int index = 0;
            for (int i = 0; i < border; ++i) {
                if (nums[i + 1] < nums[i]) {
                    nums[i] ^= nums[i + 1];
                    nums[i + 1] ^= nums[i];
                    nums[i] ^= nums[i + 1];

                    flag = true;
                    index = i;
                }
            }

            if (!flag) {
                break;
            }
            border = index;
        }
    }
}

File: py/sort/array/BubbleSort.py
#!/usr/bin/env python
# -*- coding:gbk -*-

from typing import List

class BubbleSort:
    def sort(self, nums: List[int]) -> None:
        border = len(nums) - 1
        for _ in nums:
            flag, index = False, 0
            for i in range(border):
                if nums[i + 1] < nums[i]:
                    nums[i], nums[i + 1] = nums[i + 1], nums[i]
                    flag, index = True, i

            if not flag:
                break
            border = index
